src/main/java/pvs/app/service/GitlabApiService.java

```diff
@@ -0,0 +1,133 @@
+package pvs.app.service;
+
+import org.gitlab4j.api.*;
+import org.gitlab4j.api.models.Commit;
+import org.gitlab4j.api.models.CommitStats;
+import org.gitlab4j.api.models.Issue;
+import org.gitlab4j.api.models.Project;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+import pvs.app.dto.GitlabIssueDTO;
+import pvs.app.service.thread.GitlabCommitLoaderThread;
+import pvs.app.service.thread.GitlabIssueLoaderThread;
+
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+@Service
+public class GitlabApiService {
+
+    private final WebClient webClient;
+    private final GitlabCommitService gitlabCommitService;
+    private GitLabApi gitLabApi;
+    private String username = null;
+    private List<Commit> commits;
+    private boolean useToken = true;
+
+    public GitlabApiService(WebClient.Builder webClientBuilder, GitlabCommitService gitlabCommitService) {
+        String token = System.getenv("PVS_GITLAB_TOKEN");
+        if (useToken){
+            this.gitLabApi = new GitLabApi("https://gitlab.com", Constants.TokenType.ACCESS, token);
+        }
+
+        this.gitlabCommitService = gitlabCommitService;
+        this.webClient = webClientBuilder.baseUrl("https://api.gitlab.com/")
+                .defaultHeader("Authorization", "Bearer " + token)
+                .build();
+    }
+
+    private String dateToISO8601(Date date) {
+        SimpleDateFormat sdf;
+        sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'");
+        sdf.setTimeZone(TimeZone.getTimeZone("GMT+8"));
+        return sdf.format(date);
+    }
+
+    public void oauth2login(String username, String password) throws GitLabApiException {
+        if (this.gitLabApi == null) {
+            this.gitLabApi = GitLabApi.oauth2Login("https://gitlab.com/", username, password);
+        }
+        System.out.println("oauth2login has been executed");
+        // two problems need to be dealt with:
+        // 1. password security
+        // 2. whether we need to log in whenever the service start
+    }
+
+    public Boolean getCommitsFromGitlab(String owner, String name) throws GitLabApiException, InterruptedException {
+        CommitsApi commitsApi = this.gitLabApi.getCommitsApi();
+        ProjectApi projectApi = this.gitLabApi.getProjectApi();
+        System.out.println(owner + " " + name);
+        Project project = projectApi.getProject(owner, name);
+        System.out.println("test1");
+        commits = commitsApi.getCommits(project.getId());
+        if (commits.size() != 0){
```

Comment:
The zero commit is actually a normal situation for a repository.

So suggest not use `size() != 0` in this place.